### DISCUSSION TITLE
Bad sound quality when using anything else than default audio device

### DISCUSSION THREAD

**Discussion by iko3** [mikebrady/shairport-sync#1832] [Q&A]
Hi, I installed Shairport-sync in docker container and configured it to use alsa "default" output device. 
I have one physical USB-sound card which is listed under different names when I run `aplay -L `
When I specify any other name for **same** physical output device, sound quality gets really bad and I hear lots of crackling.

The reason why I don't want to specify my device as "default" is because I'm planning to create two instances of Shairport in different containers and output sound through two different physical devices and then I can't refer to both as "default".

Here is output from `aplay -L`:

> null
    Discard all samples (playback) or generate zero samples (capture)
default
    Default Audio Device
sysdefault
    Default Audio Device
lavrate
    Rate Converter Plugin Using Libav/FFmpeg Library
samplerate
    Rate Converter Plugin Using Samplerate Library
speexrate
    Rate Converter Plugin Using Speex Resampler
jack
    JACK Audio Connection Kit
oss
    Open Sound System
pulse
    PulseAudio Sound Server
speex
    Plugin using Speex DSP (resample, agc, denoise, echo, dereverb)
upmix
    Plugin for channel upmix (4,6,8)
vdownmix
    Plugin for channel downmix (stereo) with a simple spacialization
hw:CARD=S3,DEV=0
    Sound Blaster Play! 3, USB Audio
    Direct hardware device without any conversions
plughw:CARD=S3,DEV=0
    Sound Blaster Play! 3, USB Audio
    Hardware device with all software conversions
default:CARD=S3
    Sound Blaster Play! 3, USB Audio
    Default Audio Device
sysdefault:CARD=S3
    Sound Blaster Play! 3, USB Audio
    Default Audio Device
front:CARD=S3,DEV=0
    Sound Blaster Play! 3, USB Audio
    Front output / input
surround21:CARD=S3,DEV=0
    Sound Blaster Play! 3, USB Audio
    2.1 Surround output to Front and Subwoofer speakers
surround40:CARD=S3,DEV=0
    Sound Blaster Play! 3, USB Audio
    4.0 Surround output to Front and Rear speakers
surround41:CARD=S3,DEV=0
    Sound Blaster Play! 3, USB Audio
    4.1 Surround output to Front, Rear and Subwoofer speakers
surround50:CARD=S3,DEV=0
    Sound Blaster Play! 3, USB Audio
    5.0 Surround output to Front, Center and Rear speakers
surround51:CARD=S3,DEV=0
    Sound Blaster Play! 3, USB Audio
    5.1 Surround output to Front, Center, Rear and Subwoofer speakers
surround71:CARD=S3,DEV=0
    Sound Blaster Play! 3, USB Audio
    7.1 Surround output to Front, Center, Side, Rear and Woofer speakers
iec958:CARD=S3,DEV=0
    Sound Blaster Play! 3, USB Audio
    IEC958 (S/PDIF) Digital Audio Output
dmix:CARD=S3,DEV=0
    Sound Blaster Play! 3, USB Audio
    Direct sample mixing device
usbstream:CARD=S3
    Sound Blaster Play! 3
    USB Stream Output

here is my config file:
> general =
{
	name = "Bathroom";
	port = 6001;
};
alsa =
{
	output_device = "default";
};

**Comment by mikebrady** [comment]
Thanks for the post, but this is probably best raised as an Issue. If you'd be kind enough to [open one](https://github.com/mikebrady/shairport-sync/issues/new/choose), we can try to deal with it there.

**Comment by iko3** [comment]
created issue #1833
